src/components/ManageWallets/ManageWallets.tsx

```diff
@@ -0,0 +1,90 @@
+import Button from 'components/core/Button/Button';
+import colors from 'components/core/colors';
+import Spacer from 'components/core/Spacer/Spacer';
+import ErrorText from 'components/core/Text/ErrorText';
+import { BodyMedium, BodyRegular } from 'components/core/Text/Text';
+import { USER_SIGNIN_ADDRESS_LOCAL_STORAGE_KEY } from 'contexts/auth/constants';
+import useFetcher from 'contexts/swr/useFetcher';
+import { useAuthenticatedUserAddresses } from 'hooks/api/users/useUser';
+import useAddWalletModal from 'hooks/useAddWalletModal';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { getLocalStorageItem } from 'utils/localStorage';
+import ManageWalletsRow from './ManageWalletsRow';
+
+type Props = {
+  newAddress?: string;
+};
+
+function ManageWallets({ newAddress }: Props) {
+  const addresses = useAuthenticatedUserAddresses();
+  const fetcher = useFetcher();
+
+  const [localAddresses, setLocalAddresses] = useState(addresses);
```

Comment:
handle this with optimistic update / mutation

Reply:
done ✅